src/cryptonote_basic/cryptonote_basic.h

```diff
@@ -479,8 +503,11 @@ namespace cryptonote
 
   public:
     block(): block_header(), hash_valid(false) {}
-    block(const block &b): block_header(b), hash_valid(false), miner_tx(b.miner_tx), tx_hashes(b.tx_hashes) { if (b.is_hash_valid()) { hash = b.hash; set_hash_valid(true); } }
-    block &operator=(const block &b) { block_header::operator=(b); hash_valid = false; miner_tx = b.miner_tx; tx_hashes = b.tx_hashes; if (b.is_hash_valid()) { hash = b.hash; set_hash_valid(true); } return *this; }
+
+    block(const block &b): block_header(b), hash_valid(b.is_hash_valid()), miner_tx(b.miner_tx), tx_hashes(b.tx_hashes), hash(b.hash) {}
+    block(block &&b): block_header(std::move(b)), hash_valid(b.is_hash_valid()), miner_tx(std::move(b.miner_tx)), tx_hashes(std::move(b.tx_hashes)), hash(std::move(b.hash)) { b.miner_tx.set_null(); b.tx_hashes.clear(); }
+    block &operator=(const block &b) { if(this != std::addressof(b)) { block_header::operator=(b); hash_valid = b.is_hash_valid(); miner_tx = b.miner_tx; tx_hashes = b.tx_hashes; hash = b.hash; } return *this; }
```

Comment:
NIT: Looks a little unkempt when it's one line. I know that it was like that before, but I think now would be a good time to expand it. In other words:

```cpp
Method()
{
    // Stuff
}
```
instead of
```cpp
Method() { /* Stuff */ }
```

Reply:
Updated.